editoast/src/views/infra/routes.rs

```diff
@@ -97,9 +98,15 @@ async fn get_routes_from_waypoint(
 }
 
 #[derive(Debug, Clone, Serialize, PartialEq, ToSchema)]
-#[serde(deny_unknown_fields, tag = "type", content = "track_ranges")]
+#[serde(deny_unknown_fields, tag = "type")]
 enum RouteTrackRangesResult {
-    Computed(Vec<DirectionalTrackRange>),
+    Computed {
+        track_ranges: Vec<DirectionalTrackRange>,
+        // ToSchema isn’t derived correctly with tuples (inline doesn’t work)
+        // so we need to specify the value_type
+        #[schema(inline, value_type = Vec<(String, String)>)]
+        switches_directions: Vec<(Identifier, Identifier)>,
+    },
```

Comment:
Not blocking ofc but that eliminates some duplication:
```suggestion
    Computed(RoutePath),
```